File: TodoJava/app/src/main/java/com/megait/todojava/TodoRecyclerViewAdapter.java
package com.megait.todojava;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;


@AllArgsConstructor
public class TodoRecyclerViewAdapter
        extends RecyclerView.Adapter<TodoRecyclerViewAdapter.MyViewHolder> {

    private List<Todo> list;

    public void updateList(List<Todo> list){
        this.list = list;
        notifyDataSetChanged(); // 요걸 실행해야 Adapter가 아이템의 변경을 감지하고
                                // 뷰를 업데이트한다.
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                        .inflate(R.layout.item, parent, false);
        return new MyViewHolder(view);
    }



    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.getMCheckbox().setText(list.get(position).getContent());

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    @Getter @Setter
    public class MyViewHolder extends RecyclerView.ViewHolder{
        private CheckBox mCheckbox;
        public MyViewHolder(View view){
            super(view);
            mCheckbox = view.findViewById(R.id.cb_complete);
        }
    }
}

File: TodoKotlin/app/src/main/java/com/example/todokotlin/TodoRecyclerViewAdapter.kt
package com.example.todokotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import androidx.lifecycle.LiveData
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.todojava.Todo

class TodoRecyclerViewAdapter :
    RecyclerView.Adapter<TodoRecyclerViewAdapter.TodoViewHolder>() {
    lateinit var mTodoList: List<Todo>

    fun updateList(list:List<Todo>){
        mTodoList = list
        notifyDataSetChanged()
    }

    //alt+insert -> implement Method
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TodoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item,parent,false)
        return TodoViewHolder(view)
    }

    override fun getItemCount(): Int {
        TODO("Not yet implemented")
    }

    override fun onBindViewHolder(holder: TodoViewHolder, position: Int) {
       holder.checkBox.text = mTodoList[position].content
    }

    class TodoViewHolder(var rootView: View) : RecyclerView.ViewHolder(rootView) {
        var checkBox : CheckBox
        init {
            checkBox = rootView.findViewById(R.id.cb_complete)
        }
   }
}